rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll

```diff
@@ -55,16 +55,14 @@ module SsaInput implements SsaImplCommon::InputSig<Location> {
    */
   class SourceVariable extends Variable {
     SourceVariable() {
-      this.isImmutable()
-      or
-      this.isMutable() and
-      forall(VariableAccess va | va = this.getAnAccess() |
-        va instanceof VariableReadAccess and
+      this.isMutable()
+      implies
+      not exists(VariableAccess va | va = this.getAnAccess() |
+        exists(RefExpr re | va = re.getExpr() and re.isMut())
```

Comment:
Nit: I prefer `va = any(RefExpr re | re.isMut()).getExpr()`

Reply:
Thanks. That is definitely nicer.